File: python/use_vertical_formatting.py
# Use Vertical Formatting
# You should read your code from top to bottom.
# You should read your code without "jumping" over functions.
# Similar and dependent functions should be close vertically.

# Before
class Calculator:
    def add(self, a, b):
        return a + b

    def multiply(self, a, b):
        return a * b

    def calculate(self, a, b):
        sum_result = self.add(a, b)
        difference = self.subtract(a, b)
        product = self.multiply(a, b)
        quotient = self.divide(a, b)
        return [sum_result, difference, product, quotient]

    def divide(self, a, b):
        return a / b

    def subtract(self, a, b):
        return a - b

calculator = Calculator()
result = calculator.calculate(10, 5)
print(result)  # [15, 5, 50, 2.0]


# After
class Calculator:
    def calculate(self, a, b):
        sum_result = self.add(a, b)
        difference = self.subtract(a, b)
        product = self.multiply(a, b)
        quotient = self.divide(a, b)
        return [sum_result, difference, product, quotient]

    def add(self, a, b):
        return a + b

    def subtract(self, a, b):
        return a - b

    def multiply(self, a, b):
        return a * b

    def divide(self, a, b):
        return a / b

calculator = Calculator()
result = calculator.calculate(10, 5)
print(result)  # [15, 5, 50, 2.0]

File: csharp/UseVerticalFormatting.cs
// Use Vertical Formatting
// You should read your code from top to bottom.
// You should read your code without "jumping" over functions.
// Similar and dependent functions should be close vertically.

using System;

public class UseVerticalFormatting
{
    // Before
    public class CalculatorBefore
    {
        public int Add(int a, int b)
        {
            return a + b;
        }

        public int Multiply(int a, int b)
        {
            return a * b;
        }

        public int[] Calculate(int a, int b)
        {
            int sumResult = Add(a, b);
            int difference = Subtract(a, b);
            int product = Multiply(a, b);
            int quotient = Divide(a, b);
            return new int[] { sumResult, difference, product, quotient };
        }

        public int Divide(int a, int b)
        {
            return a / b;
        }

        public int Subtract(int a, int b)
        {
            return a - b;
        }
    }

    // After
    public class Calculator
    {
        public int[] Calculate(int a, int b)
        {
            int sumResult = Add(a, b);
            int difference = Subtract(a, b);
            int product = Multiply(a, b);
            int quotient = Divide(a, b);
            return new int[] { sumResult, difference, product, quotient };
        }

        public int Add(int a, int b)
        {
            return a + b;
        }

        public int Subtract(int a, int b)
        {
            return a - b;
        }

        public int Multiply(int a, int b)
        {
            return a * b;
        }

        public int Divide(int a, int b)
        {
            return a / b;
        }
    }

    public static void Main(string[] args)
    {
        var calculatorBefore = new CalculatorBefore();
        int[] resultBefore = calculatorBefore.Calculate(10, 5);
        Console.WriteLine($"[{string.Join(", ", resultBefore)}]"); // [15, 5, 50, 2]

        Console.WriteLine();

        var calculator = new Calculator();
        int[] result = calculator.Calculate(10, 5);
        Console.WriteLine($"[{string.Join(", ", result)}]"); // [15, 5, 50, 2]
    }
}
